posthog/tasks/scheduled.py

```diff
@@ -130,6 +132,14 @@ def setup_periodic_tasks(sender: Celery, **kwargs: Any) -> None:
         name="update quota limiting",
     )
 
+    # Send all periodic digest reports
+    weekly_digest_end_date = datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
```

Comment:
The `weekly_digest_end_date` will be evaluated when the code is first loaded/instantiated, not when the task runs. This means the datetime will be fixed to whatever time the code was initialized, which is probably not what you want for a weekly report.

You could move the datetime calculation inside the function:
```
def send_all_periodic_digest_reports(end_date=None):
    if end_date is None:
        end_date = datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
    # rest of your report logic...

sender.add_periodic_task(
    crontab(hour="9", minute="0", day_of_week="mon"),
    send_all_periodic_digest_reports.s(),  # Remove the end_date parameter
    name="send all weekly digest reports",
)
```

Reply:
I think this is the only blocker from my comments.

Reply:
I think I do want to define the time when it's called. That way we can be flexible with the date. Say the task fails and we want to resend it on a later date, but still only get the updates for the prior week. We need to define the end date at the beginning in that case. Similar to usage reports.

Reply:
No but this won't work because the date time will be fixed to the time the server started so it'll get reset each time there is a deployment. My suggested change would make it so it uses the time when the function is called based on the cron.

Reply:
OH

Reply:
The date is still flexible with an optional end date arg

Reply:
Yeah sorry I was just being dumb, good call-out